metaflow/datastore/content_addressed_store.py

```diff
@@ -29,18 +29,19 @@ def __init__(self, flow_datastore, prefix):
 
         Parameters
         ----------
-        flow_datastore : FlowDataStore
-            Flow-level datastore (parent of this and any TaskDataStore)
         prefix : string
             "Directory" that will be prepended when storing a file
+        backend_class : type
+            Class for the backing DataStoreBackend to use; if not provided use
+            default_backend_class
         """
         self._prefix = prefix
-        self._backend = flow_datastore._backend
-        self._logger = flow_datastore.logger
-        self._monitor = flow_datastore.monitor
-        self._blob_cache = flow_datastore.blob_cache
-
+        self._backend = backend
         self.TYPE = self._backend.TYPE
+        self._blob_cache = None
+
+    def set_blob_cache(self, blob_cache):
```

Comment:
Nit, let's use a property here (setter and getter).

Reply:
is it a better approach? `@properties` are nice for read-only getters but I find it better to use explicit functions for mutations

Reply:
OK, no huge preference.